src/modules/auth/join/JoinPage.vue

```diff
@@ -149,6 +149,12 @@ onBeforeMount(async () => {
     ? Number(route.query.amount)
     : joinContent.value.initialAmount;
 
+  signUpData.period = Object.values(ContributionPeriod).includes(
+    route.query.period
+  )
+    ? (route.query.period as ContributionPeriod)
+    : ContributionPeriod.Monthly;
```

Comment:
Setting this instead of a default value as it's what we're doing above in the very definition of `signUpData`.

Reply:
Hmm strange I think that might be a bug as it should be getting set to `joinContent.value.initialPeriod` once `joinContent` has loaded, could you change it?